lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1071,25 +1070,27 @@ def get_issued_certificates(
         Returns:
             dict: Certificates per application name.
         """
-        certificates: Dict[str, Dict[str, str]] = defaultdict(dict)
+        certificates: Dict[str, List[Dict[str, str]]] = {}
         relations = (
-            [
-                relation
-                for relation in self.model.relations[self.relationship_name]
-                if relation.id == relation_id
-            ]
-            if relation_id is not None
+            [self.model.relations[self.relationship_name][relation_id]]
```

Comment:
I think `self.model.relations[self.relationship_name]` will return a list of relations, not a dictionary, so assuming it has a `[relation_id]` key is incorrect.
At least that's what I understood from the documentation.
https://ops.readthedocs.io/en/latest/#ops.Model.relations

Reply:
It looks like this was changed recently at commit a013811 after I made my own branch but before I made my first commit, so I overwrote it without noticing. I have reverted it.

Reply:
It doesn't fail because when you create a relation with harness in the tests, it will give 0 as relation id to the first relation, then 1, then 2 and so on, which then can be used as indices without an issue.
One way to test is to call `get_issued_certificates` from tests with a non existing relation id and see that:
1. it does not return a relation.
2. it does not error.
A minimal example would be something like:
```
def test_given_incorrect_relation_id_when_get_issued_certificates_by_relation_id_then_returned_dict_is_empty(
        self,
    ):
        random_relation_id = 1234
        self.harness.set_leader(is_leader=True)
        certificates = self.harness.charm.certificates.get_issued_certificates(random_relation_id)
        self.assertEqual(certificates, {})
```

This would fail with an error in the first case, but now it should pass.

Reply:
Nice, thanks :)